clients/da-vinci-client/src/main/java/com/linkedin/davinci/blobtransfer/P2PBlobTransferManagerFactory.java

```diff
@@ -0,0 +1,183 @@
+              globalTrafficHandler),
+          new NettyFileTransferClient(
+              blobTransferConfig.getP2pTransferClientPort(),
+              blobTransferConfig.getBaseDir(),
+              storageMetadataService,
+              blobTransferConfig.getPeersConnectivityFreshnessInSeconds(),
+              globalTrafficHandler),
+          blobFinder,
+          blobTransferConfig.getBaseDir(),
+          aggVersionedBlobTransferStats);
+
+      manager.start();
+      return manager;
+    } catch (Exception e) {
+      // swallow the exception and continue the consumption via pubsub system
+      LOGGER.warn("Failed to start up the P2P blob transfer manager", e);
+      return null;
+    }
+  }
+
+  /**
+   * Builder for {@link P2PBlobTransferManagerFactory} to create a new instance.
+   */
+  public static class P2PBlobTransferManagerFactoryBuilder {
+    private P2PBlobTransferConfig blobTransferConfig;
+    private ClientConfig clientConfig;
+    private CompletableFuture<HelixCustomizedViewOfflinePushRepository> customizedViewFuture;
+    private StorageMetadataService storageMetadataService;
+    private ReadOnlyStoreRepository readOnlyStoreRepository;
+    private StorageEngineRepository storageEngineRepository;
+    private AggVersionedBlobTransferStats aggVersionedBlobTransferStats;
+
+    public P2PBlobTransferManagerFactoryBuilder setBlobTransferConfig(P2PBlobTransferConfig blobTransferConfig) {
+      this.blobTransferConfig = blobTransferConfig;
+      return this;
+    }
+
+    public P2PBlobTransferManagerFactoryBuilder setClientConfig(ClientConfig clientConfig) {
+      this.clientConfig = clientConfig;
+      return this;
+    }
+
+    public P2PBlobTransferManagerFactoryBuilder setCustomizedViewFuture(
+        CompletableFuture<HelixCustomizedViewOfflinePushRepository> customizedViewFuture) {
+      this.customizedViewFuture = customizedViewFuture;
+      return this;
+    }
+
+    public P2PBlobTransferManagerFactoryBuilder setStorageMetadataService(
+        StorageMetadataService storageMetadataService) {
+      this.storageMetadataService = storageMetadataService;
+      return this;
+    }
+
+    public P2PBlobTransferManagerFactoryBuilder setReadOnlyStoreRepository(
+        ReadOnlyStoreRepository readOnlyStoreRepository) {
+      this.readOnlyStoreRepository = readOnlyStoreRepository;
+      return this;
+    }
+
+    public P2PBlobTransferManagerFactoryBuilder setStorageEngineRepository(
+        StorageEngineRepository storageEngineRepository) {
+      this.storageEngineRepository = storageEngineRepository;
+      return this;
+    }
+
+    public P2PBlobTransferManagerFactoryBuilder setAggVersionedBlobTransferStats(
+        AggVersionedBlobTransferStats aggVersionedBlobTransferStats) {
+      this.aggVersionedBlobTransferStats = aggVersionedBlobTransferStats;
+      return this;
+    }
+
+    public P2PBlobTransferManagerFactory build() {
+      return new P2PBlobTransferManagerFactory(
```

Comment:
In theory, we can pass `builder` as a param to `P2PBlobTransferManagerFactory` constructor to avoid the long param list.